### ISSUE TITLE
[RFC] Decide officially supported/tested platform

### ISSUE THREAD

**Issue by YJDoc2** [containers/youki#2038]
This issue originates from [this](https://github.com/containers/youki/pull/2012#issuecomment-1579244235) and [this](https://github.com/containers/youki/pull/2012#issuecomment-1579930218). Particularly the comment 

>Here is what I propose. We declare that Ubuntu and Fedora are the official dev environment we support. We will create automation for setting up the environment for these two OS. Eventually, we would even make CI a matrix strategy running both Ubuntu and Fedora.

Given that this would be a big commitment as well as a big decision as a project, opening this RFC for discussion.

Note that this was already once considered in https://github.com/containers/youki/issues/39 and in even more detail in https://github.com/containers/youki/issues/369.

**Comment by utam0k** [comment]
I consider the cost of managing these to be high. And I do not think there is any part of Linux distros that makes a difference.
I want to support architectures other than x86, but I don't realistically have a CI to run them.
Would anyone actually benefit from doing this tier? I fear these will be a strong drag on the maintainer. We do not want to strangle ourselves.

**Comment by yihuaf** [comment]
I agree with @utam0k that the cost to maintain many distro/targets can be high. The project is best served to focus one, at most two platforms/distros/targets and build `youki` to be rock solid. Today, the only implicit "official" support we declare is Ubuntu x86 and kernel version >= 5.3, since we run CI tests on this target. We also do provide instructions for `fedora and family`, but we don't actively test them in CI. I mention fedora here only because it is one of the easiest way to run kernel that is closer to the bleeding edge, and it is likely the 2nd most popular dev platform in the wild. At the same time, github CI doesn't support fedora last time I checked.

For all other platforms and architectures, I would default to "Not for Now".

**Comment by yihuaf** [comment]
Oh, and in addition to platform, we may also want to support `musl` as a build target. There are two usecases:

1. statically linked youki binary. (libseccomp and libdbus is a blocker)
2. projects using `libcontainer` such as aurae, whose maintainer implemented the first `musl` support for us.
